app/decoding/communities/[communityId]/page.tsx

```diff
@@ -0,0 +1,1451 @@
+                                        `}
+                                        disabled={
+                                            week > getCurrentWeek(communityData)
+                                        }
+                                    >
+                                        {week}
+                                    </button>
+                                ))}
+                            </div>
+                            {selectedWeek === getCurrentWeek(communityData) && (
+                                <span className="ml-2 px-2 py-0.5 text-xs rounded-full bg-orange-100 dark:bg-orange-500/20 text-orange-600 dark:text-orange-400 whitespace-nowrap">
+                                    Current
+                                </span>
+                            )}
+                        </div>
+                    </div>
+                </div>
+            </div>
+
+            {/* Main Content */}
+            <div className="max-w-7xl mx-auto px-4 py-6">
+                <div className="grid grid-cols-1 md:grid-cols-12 gap-6">
+                    {/* Sidebar with Desktop Week Navigation */}
+                    <div className="md:col-span-3">
+                        <div className="space-y-4">
+                            <Progress community={communityData} />
+                            <WeeklyCallSection weekData={currentWeekData} />
+                            {/* Desktop Week Navigation */}
+                            <div className="hidden md:block">
+                                <div className="bg-white dark:bg-gray-800/40 rounded-xl overflow-hidden border border-gray-200 dark:border-gray-800">
+                                    <div className="p-4 border-b border-gray-200 dark:border-gray-800">
+                                        <h3 className="text-sm font-medium text-gray-900 dark:text-white">
+                                            Course Timeline
+                                        </h3>
+                                    </div>
+                                    <div className="divide-y divide-gray-200 dark:divide-gray-700">
+                                        {Array.from(
+                                            { length: totalWeeks },
+                                            (_, i) => i + 1
+                                        ).map((week) => {
+                                            const isLocked =
+                                                week >
+                                                getCurrentWeek(communityData)
+                                            const isPast =
+                                                week <
+                                                getCurrentWeek(communityData)
+                                            const isCurrent =
+                                                week ===
+                                                getCurrentWeek(communityData)
```

Comment:
`getCurrentWeek` is called multiple times
we can initialise it higher up and use that data when we need it